src/gravel/controllers/resources/network.py

```diff
@@ -0,0 +1,305 @@
+        return self._interfaces
+
+    @property
+    def nameservers(self) -> List[str]:
+        return self._nameservers
+
+    @property
+    def routes(self) -> List[RouteModel]:
+        return self._routes
+
+    async def _do_tick(self) -> None:
+        self._refresh_interfaces()
+        self._refresh_nameservers()
+        self._refresh_routes()
+        logger.debug(self._interfaces)
+        logger.debug(self._nameservers)
+        logger.debug(self._routes)
+
+    async def _should_tick(self) -> bool:
+        return not self._is_busy
+
+    def _refresh_interfaces(self) -> None:
+        # interfaces is the union of the physical interfaces (/sys/class/net/...) and
+        # any interfaces for which we have config files (/etc/sysconfig/network/ifcfg-*)
+        interfaces: Dict[str, InterfaceModel] = {}
+
+        # This gets the physical interfaces
+        sys_class_net = Path("/sys/class/net")
+        assert sys_class_net.exists() and sys_class_net.is_dir()
+        for dev in sys_class_net.glob("*"):
+            device = dev.joinpath("device")
+            if device.exists():
+                # /sys/class/net/$dev/device exist, so it's a physical interface
+                interfaces[dev.stem] = InterfaceModel(name=dev.stem)
+
+        # ...and this gets any configured interfaces
+        sysconfig = Path("/etc/sysconfig/network")
+        assert sysconfig.exists() and sysconfig.is_dir()
+        for conf in sysconfig.glob("ifcfg-*"):
+            # Blacklist here is lifted from wicked source
+            if conf.name.endswith("~") or conf.suffix in [
+                ".old",
+                ".bak",
+                ".orig",
+                ".scpmbackup",
+                ".rpmnew",
+                ".rpmsave",
+                ".rpmorig",
+            ]:
+                logger.debug(f"Skipping blacklisted suffix on {conf}")
+                continue
+            iface = conf.stem[6:]
+            if iface == "lo":
+                # We don't care about the loopback interface
+                continue
+            logger.debug(f"Reading {iface} config from {conf}")
+
+            rawconf = self._parse_config(conf)
+
+            config = InterfaceConfigModel()
+            if "IPADDR" in rawconf:
+                rawaddr = rawconf["IPADDR"]
+                if rawaddr:
+                    config.ipaddr = rawaddr
+                    # TODO: do we want to separate netmask etc. out?
+                    # (it can currently only be specified as "/" suffix
+                    # on ipaddr)
+            if "BOOTPROTO" in rawconf and rawconf["BOOTPROTO"] is not None:
+                config.bootproto = rawconf["BOOTPROTO"]
+
+            if iface not in interfaces:
+                interfaces[iface] = InterfaceModel(name=iface)
+            interfaces[iface].config = config
+
+        self._interfaces = interfaces
+
+    def _refresh_nameservers(self) -> None:
+        config = Path("/etc/sysconfig/network/config")
```

Comment:
same question vs using other utilities (or /etc/resolv.conf etc).

Reply:
With the default setup, /etc/resolv.conf is dynamically generated by netconfig based on variables in /etc/sysconfig/network/config, so it should be OK like this, provided my various assumptions are correct - see my other comment on that.